pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -435,30 +448,118 @@ contract ManagedPool is BaseWeightedPool, ReentrancyGuard {
         SwapRequest memory swapRequest,
         uint256 currentBalanceTokenIn,
         uint256 currentBalanceTokenOut
-    ) internal override returns (uint256) {
+    ) internal virtual override whenNotPaused returns (uint256) {
+        // Swaps are disabled while the contract is paused.
```

Comment:
The `whenNotPaused` modifier is already applied by BaseWeightedPool, we shouldn't repeat it here